### ISSUE TITLE
An array of size of (a particular array's length + 1) is not recognised as @MinLen(1)

### ISSUE THREAD

**Issue by Kakarot-SSJ4** [typetools/checker-framework#2632]
Consider the following code:

```
import org.checkerframework.common.value.qual.MinLen;
class tryit{
	void issue(final int[] a){
		int @MinLen(1) [] b = new int[a.length + 1];
	}
}
```

Run it by: `javac -processor index tryit.java`

Output:
```
tryit.java:4: error: [assignment.type.incompatible] incompatible types in assignment.
		int @MinLen(1) [] b = new int[a.length + 1];
		                      ^
  found   : @UnknownVal int @ArrayLenRange(from=0, to=2147483647) []
  required: @UnknownVal int @ArrayLenRange(from=1, to=2147483647) []
1 error
```

However, an array's length >= 0, hence `a.length + 1` should be recognised as greater than or equal to 1.

**Comment by Prajwal7842** [comment]
@kelloggm @Kakarot-SSJ4  Is this issue still valid? If yes, I would like to work on it.

**Comment by kelloggm** [comment]
You can check whether it is valid by running the provided test case through the Index Checker. Does it still issue the false positive warning? (I suspect it does - if we had fixed this, then we would have closed the issue)

**Comment by Prajwal7842** [comment]
@kelloggm 
I tried creating this issue again. It seems this issue has been fixed and checker framework does not give any warnings now. This issue can be closed.
Also, could you suggest me some easy issues that I can start working on or any other feature or test that I can add.

**Comment by kelloggm** [comment]
If the issue can be closed, you should write a pull request with just the test case that demonstrates that the test case passes. We can't close the issue until the test case has been added to the test suite.

While you're doing so, please give the test case a sensible name that follows the standard Java conventions (i.e. begins with a capital letter, and describes the problem that is being tested).

Thanks!

**Comment by Prajwal7842** [comment]
@kelloggm 
**This issue is still valid**. The framework gives the current errors as presented in the issue above.
Apologise for the above comment.
I ran the initial tests on Live Demo Page and it didn't gave me any error.
When I tested the same on my local computer, the warnings were given by the framework.
 
I would like to work on this issue. 
Also, could you just help me from where should I start to solve this.

**Comment by wmdietl** [comment]
> I ran the initial tests on Live Demo Page and it didn't gave me any error.

Maybe you didn't select the Index Checker?
The live demo gives the expected error for that [example](http://eisop.uwaterloo.ca/live#mode=display&code=import+org.checkerframework.common.value.qual.MinLen%3B%0D%0Aclass+tryit%7B%0D%0A%09void+issue(final+int%5B%5D+a)%7B%0D%0A%09%09int+%40MinLen(1)+%5B%5D+b+%3D+new+int%5Ba.length+%2B+1%5D%3B%0D%0A%09%7D%0D%0A%7D&typeSystem=index).

**Comment by Prajwal7842** [comment]
Yeah I think so. But how can we correct this error and ensure warning is not displayed?

**Comment by kelloggm** [comment]
By ensuring that the type of the expression `new int[a.length + 1]` is `@MinLen(1)`, probably by modifying the implementation of the Value Checker. That's a programming task (and I encourage you to try it!).

**Comment by Prajwal7842** [comment]
I will surely try to fix the issue. Thanks for the help.

**Comment by kelloggm** [comment]
@MrigankArora and I discussed this issue in detail. We believe it is valid; the error mentioned above is caused by the Value Checker's support for overflow. Adding 1 to `a.length` could possibly cause an overflow, so the type of `a.length + 1` is `@UnknownVal int`, which is correct in this case.

There are two possible solutions:
* annotate `a` as `@ArrayLenRange(from=0, to=2147483646)` (should we have syntactic sugar for this?)
* use the Value Checker's ignore overflow flag
